routemaster/jobs/batch.rb

```diff
@@ -35,21 +36,15 @@ def call(uid)
           batch.delete
         rescue Services::Deliver::CantDeliver => e
           _log_exception(e)
-          attempts = batch.fail
-          raise Models::Queue::Retry, _backoff(attempts)
+          raise Models::Queue::Retry, _backoff(batch)
         end
         self
       end
 
       private
       
-      def _backoff(attempts)
-        backoff = 1_000 * 2 ** [attempts-1, _backoff_limit].min
-        backoff + rand(backoff)
-      end
-
-      def _backoff_limit
-        @@_backoff_limit ||= Integer(ENV.fetch('ROUTEMASTER_BACKOFF_LIMIT'))
+      def _backoff(batch)
+        Services::Throttle.new(batch: batch).retry_backoff
```

Comment:
the `batch` and `subscriber` models here should be "hot", and already have the attributes in memory. The reason I'm re-initializing it here is that it needs a reference to the original batch to calculate the "legacy" backoff.

I could dump that to make it simpler, I think, but I think there is some value in being able to support both, at least while we test the new backoff system and we're sure we're happy with it.

I could pass the batch to `Services::Deliver`, and move the extraction of the events (the `@buffer`) in there.